src/docker/ContextManager.ts

```diff
@@ -218,21 +220,37 @@ export class DockerContextManager implements ContextManager, Disposable {
 
             // Setting the DOCKER_CONTEXT environment variable to whatever is passed along means the CLI will always
             // return that specified context as Current = true. This way we don't need extra logic below in parsing.
+            // TODO: eventually change to `docker context ls --format json`
             const { stdout } = await execAsync('docker context ls --format="{{json .}}"', { ...ContextCmdExecOptions, env: { ...process.env, DOCKER_CONTEXT: dockerContextEnv } });
-            const lines = stdout.split(/\r?\n/im);
 
-            for (const line of lines) {
-                // Blank lines should be skipped
-                if (!line) {
-                    continue;
+            try {
+                // Try parsing as-is; newer CLIs output a JSON object array
+                const contexts = JSON.parse(stdout) as DockerContext[];
+
+                result.push(...contexts.map(context => {
+                    return {

```

Comment:
Perhaps consolidate conversion of `DockerContext` objects to avoid duplication (such as the logic for setting `ContextType`)?